### ISSUE TITLE
UnobservedTaskException's caused by Smb2Client.Disconnect method

### ISSUE THREAD

**Issue by raufismayilov** [TalAloni/SMBLibrary#256]
Calling Smb2Client.Disconnect causes UnobservedTaskExceptions. This happens even if just Smb2Client.Connect and then Smb2Client.Disconnect is called.
On Linux the error is as follows:
```
System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (Operation canceled)
 ---> System.Net.Sockets.SocketException (125): Operation canceled
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Threading.Tasks.ValueTask`1.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)
   --- End of inner exception stack trace ---
```
On Windows the error is
`System.Net.Sockets.SocketException (995): The I/O operation has been aborted because of either a thread exit or an application request.`

This is most likely caused by the fact that when the socket is closed Socket.EndReceive is not called in OnClientSockerReceived.
The same applies to Smb1Client as well

**Comment by raufismayilov** [comment]
This PR fixes this issue.

https://github.com/TalAloni/SMBLibrary/pull/258

**Comment by TalAloni** [comment]
Thank you Rauf for working on this issue.

**Comment by TalAloni** [comment]
> This is most likely caused by the fact that when the socket is closed Socket.EndReceive is not called in OnClientSockerReceived. The same applies to Smb1Client as well

Can this be because Socket.Close is not called in OnClientSockerReceived?
I appreciate this theory, but a more concrete analysis will be much more helpful.

**Comment by raufismayilov** [comment]
You can ensure that Socket.EndReceive is always called and you will see that Unobserved exceptions are gone.
This is your current check that prevents from this to happen:

```
if (!clientSocket.Connected)
{
        state.ReceiveBuffer.Dispose();
        return;
}
```

**Comment by raufismayilov** [comment]
Hello TelAloni,
I will do the test and let you know, in the meantime, if you want I can set up a call and demo the issue.

**Comment by TalAloni** [comment]
Thanks, I'm waiting to hear is an issue following the latest change. if there is no issue anymore - no need to demo it.

**Comment by TalAloni** [comment]
Thank you Rauf for the testing and assistance!

**Comment by TalAloni** [comment]
This issue has been solved in v1.5.2
